core/app/models/spree/country.rb

```diff
@@ -3,6 +3,10 @@ class Country < Spree::Base
     has_many :states, dependent: :destroy
     has_many :addresses, dependent: :restrict_with_error
 
+    # we need to have this callback before any dependent: :destroy associations
+    # https://github.com/rails/rails/issues/3458
+    before_destroy :ensure_not_default
```

Comment:
@damianlegawiec  In this case It should be at above to has_many associations with, Isn't it?